pixeltable/functions/openai.py

```diff
@@ -61,28 +170,85 @@ def vision(
              }}
          ]}
     ]
-    result = env.Env.get().openai_client.chat.completions.create(
+    result = openai_client().chat.completions.create(
         messages=messages,
         model=model
     )
     return result.choices[0].message.content
 
 
-@pxt.udf
-def moderations(input: str, model: Optional[str] = None) -> dict:
-    result = env.Env().get().openai_client.moderations.create(input=input, model=model)
-    return result.dict()
-
+#####################################
+# Embeddings Endpoints
 
 @pxt.udf(batch_size=32, return_type=ts.ArrayType((None,), dtype=ts.FloatType()))
-def embeddings(input: Batch[str], *, model: str) -> Batch[np.ndarray]:
-    result = env.Env().get().openai_client.embeddings.create(
+@_retry
+def embeddings(
+        input: Batch[str],
+        *,
+        model: str,
+        user: Optional[str] = None
+) -> Batch[np.ndarray]:
+    result = openai_client().embeddings.create(
         input=input,
         model=model,
+        user=_opt(user),
         encoding_format='float'
     )
-    embeddings = [
+    return [
         np.array(data.embedding, dtype=np.float64)
         for data in result.data
     ]
-    return embeddings
+
+
+#####################################
+# Images Endpoints
+
+@pxt.udf
+@_retry
+def image_generations(
+        prompt: str,
+        *,
+        model: Optional[str] = None,
+        quality: Optional[str] = None,
+        size: Optional[str] = None,
+        style: Optional[str] = None,
+        user: Optional[str] = None
+) -> PIL.Image.Image:
```

Comment:
Do we know what the resolution is? Would be nice to return that as part of the type.